src/renderer/pages/settings-page.component.tsx

```diff
@@ -609,6 +611,45 @@ function AdvancedSettings() {
         setUseSymlink(() => newUseSymlink);
     }
 
+    const onChangeUseSystemProxy = async (newUseSystemProxy: boolean) => {
+
+        if (window.electron.platform !== "win32" || newUseSystemProxy === useSystemProxy) {
+            return;
+        }
+
+        if(!newUseSystemProxy){
+            const res = await modal.openModal(BasicModal, { data: {
+                title: "pages.settings.advanced.use-system-proxy.modal.title",
+                body: "pages.settings.advanced.use-system-proxy.modal.body",
+                image: BeatConflict,
+                buttons: [
+                    { id: "cancel", text: "misc.cancel", type: "cancel" },
+                    { id: "confirm", text: "pages.settings.advanced.use-system-proxy.modal.confirm-btn", type: "error", onClick: () => true }
+                ]
+            }});
+
+            if(res.exitCode !== ModalExitCode.COMPLETED || res.data !== "confirm"){ return; }
+        }
+
+        const { error } = await tryit(() => staticConfig.set("use-system-proxy", newUseSystemProxy));
+
+        if(error){
+            notification.notifyError({ title: "notifications.types.error", desc: "pages.settings.advanced.use-system-proxy.error-notification.message" });
+            return;
+        }
+
+        setUseSystemProxy(() => newUseSystemProxy);
+
+        if(!progressBar.require()){
+            return;
+        }
+
+        if (!newUseSystemProxy) {
+            await lastValueFrom(ipc.sendV2("restart-app"));
```

Comment:
Does the app need to be restarted in order for the change to kick in? If I'm reading how the proxy enabling works, its just setting the `process.env.GLOBAL_AGENT_HTTP/S_PROXY` right? So in theory you can just delete the keys so that the proxy is disabled. Can you double check if thats the case.

Reply:
> Does the app need to be restarted in order for the change to kick in? If I'm reading how the proxy enabling works, its just setting the `process.env.GLOBAL_AGENT_HTTP/S_PROXY` right? So in theory you can just delete the keys so that the proxy is disabled. Can you double check if thats the case.

I tried that, empty the keys do not disable the proxy agent. You may give another try to see if it works.

Reply:
Got to enable/disable toggling working without restarting the app. Read the docs that you can edit the `global.GLOBAL_AGENT` object in runtime once `bootstrap` setup is called. So I did the below code:

```typescript
// Changed `configureWindowsProxy` to `enableWindowsProxy`
async function enableWindowsProxy(enable: boolean): Promise<void> {
    if (!(await isProxyEnabled().catch(err => log.error(err)))) {
        log.info(`configureWindowsProxy: System proxy not detected`);
        return;
    }

    let { GLOBAL_AGENT: globalProxyAgent } = global as any;
    if (!globalProxyAgent) { // If this is undefined, call bootstrap to set it up
        if (!bootstrap()) {
            log.error(`configureWindowsProxy: Could not setup proxy stuff`);
            return;
        }
        globalProxyAgent = (global as any).GLOBAL_AGENT;
    }

    if (enable) {
        const httpProxyUrl = `http://${await getProxyServer().catch(err => log.error(err))}`
        globalProxyAgent.HTTP_PROXY = httpProxyUrl;
        globalProxyAgent.HTTPS_PROXY = httpProxyUrl;
        globalProxyAgent.NO_PROXY = `${await getProxyOverride().catch(err => log.error(err))}`;
    } else {
        delete globalProxyAgent.HTTP_PROXY;
        delete globalProxyAgent.HTTPS_PROXY;
        delete globalProxyAgent.NO_PROXY;
    }

    log.info(`configureWindowsProxy: Using system proxy: ${globalProxyAgent}`);
}
```

That should be the core of the change, you can handle the rest of the change at this point.

Reply:
Thanks! Done.